### DISCUSSION TITLE
Savings question

### DISCUSSION THREAD

**Discussion by diveleader** [jasonacox/Powerwall-Dashboard#244] [Q&A]
Hi,

Firstly, a big thank you to all on here for providing this excellent code.

It's been a learning curve for myself (56 years old) - re using an old N4200 laptop, installing Mint and beginning to learn Linux!

I've got Powerwall-Dashboard up and running fine, retrieved my 3+ years of Powerwall data from Tesla and it seems to be working well.

I have one question on Savings values.

My supplier in the UK gives me a 3 hour window off peak at 0.19 per kW/Hr, when I can charge my Powerwall.

Any surplus solar goes to the grid at an average credit of 0.28 per kW/Hr.

I'm playing with the savings values in the database (0.19) and I'm not sure what to set them at?

ie, what values for fp / tp and tg?  I was thinking that, since the solar is free and export to the grid is credited, they should be a minus figure?

If someone could help please, based on the following:

 - I can charge the PW offpeak at 0.19
 - Any solar produced during the day saves me from drawing grid power at 0.32
 - Any surplus generation is fed back to the grid and I'm paid 0.28

Thanks

**Comment by ThePnuts** [comment]
The dashboard as it exists does not support peak\off peak\etc and can only use a fixed price value. There is a discussion here with a few people tackling the problem, but there are many many different variations of pricing all around the world making a simple solution not possible.

For myself, I changed all of the 0.19 to 0.37 which is my cheaper rate. I wont know my true savings since I cant account for TOU, but it gives me a good idea for now.

https://github.com/jasonacox/Powerwall-Dashboard/discussions/225 and https://github.com/jasonacox/Powerwall-Dashboard/discussions/68
